lib/cxml/credential.rb

```diff
@@ -1,60 +1,54 @@
-# element. The Identity element states who the Credential represents, while the optional 
+# element. The Identity element states who the Credential represents, while the optional
 # authentication elements verify the identity of the party
 #
 # Credential has the following attributes:
-# 
+#
 # *domain*
-# Specifies the type of credential. This attribute allows 
-# documents to contain multiple types of credentials for multiple 
+# Specifies the type of credential. This attribute allows
+# documents to contain multiple types of credentials for multiple
 # authentication domains.
-# For messages sent on the Ariba Supplier Network, for 
-# instance, the domain can be AribaNetworkUserId to indicate an 
-# email address, DUNS for a D-U-N-S number, or NetworkId for a 
+# For messages sent on the Ariba Supplier Network, for
+# instance, the domain can be AribaNetworkUserId to indicate an
+# email address, DUNS for a D-U-N-S number, or NetworkId for a
 # preassigned ID.
 #
 # *type* - optional
-# Requests to or from a marketplace identify both the 
+# Requests to or from a marketplace identify both the
 # marketplace and the member company in From or To Credential
-# elements. In this case, the credential for the marketplace uses 
+# elements. In this case, the credential for the marketplace uses
 # the type attribute, which is set to the value “marketplace”
 #
 # *SharedSecred*
 # The SharedSecret element is used when the Sender has a password that the requester recognizes.
 #
 # *CredentialMac*
-# The CredentialMac element is used for the Message Authentication Code (MAC) 
-# authentication method. This authentication method is used in situations where the 
-# sender must prove to the receiver that it has been authenticated by shared secret by a 
-# trusted third party. For example, a direct PunchOut request can travel directly from a 
-# buyer to a supplier without going through a network commerce hub, because it 
-# contains a MAC (generated by the network commerce hub) that allows the supplier to 
+# The CredentialMac element is used for the Message Authentication Code (MAC)
+# authentication method. This authentication method is used in situations where the
+# sender must prove to the receiver that it has been authenticated by shared secret by a
+# trusted third party. For example, a direct PunchOut request can travel directly from a
+# buyer to a supplier without going through a network commerce hub, because it
+# contains a MAC (generated by the network commerce hub) that allows the supplier to
 # authenticate it.
 
 module CXML
   class Credential
-    attr_accessor :domain
-    attr_accessor :type
-    attr_accessor :shared_secret
-    attr_accessor :credential_mac
-    attr_accessor :identity
+    attr_reader :domain, :type, :shared_secret, :identity
 
     # Initialize a new Credential instance
     # @param data [Hash] optional initial data
     def initialize(data={})
```

Comment:
as we discussed earlier such default `(data={})` is not garantees hash - in case when `nil` specified directly
such thing may happen like:

    MyClass.new(opts[:foo])

when `opts[:foo]` is `nil`, `data` will not be hash

better to define it as `(data=nil)` to make arg optional
and then init it like `data ||= {}`